exporter/awsemfexporter/metric_translator_test.go

```diff
@@ -273,6 +273,98 @@ func TestCalculateRate(t *testing.T) {
 	assert.Equal(t, 0.5, rate)
 }
 
+func TestDimensionRollup(t *testing.T) {
+	testCases := []struct{
+		testName string
+		dimensionRollupOption string
+		dims []string
+		instrumentationLibName string
+		expected [][]string
+	}{
+		{
+			"no rollup w/o instrumentation library name",
+			"",
+			[]string{"a", "b", "c"},
+			noInstrumentationLibraryName,
+			nil,
+		},
+		{
+			"no rollup w/ instrumentation library name",
+			"",
+			[]string{"a", "b", "c"},
+			"cloudwatch-otel",
+			nil,
+		},
+		{
+			"single dim w/o instrumentation library name",
+			SingleDimensionRollupOnly,
+			[]string{"a", "b", "c"},
+			noInstrumentationLibraryName,
+			[][]string{
+				{"a"},
+				{"b"},
+				{"c"},
+			},
+		},
+		{
+			"single dim w/ instrumentation library name",
+			SingleDimensionRollupOnly,
+			[]string{"a", "b", "c"},
+			"cloudwatch-otel",
+			[][]string{
+				{OTellibDimensionKey, "a"},
+				{OTellibDimensionKey, "b"},
+				{OTellibDimensionKey, "c"},
+			},
+		},
+		{
+			"single dim w/o instrumentation library name and only one label",
+			SingleDimensionRollupOnly,
+			[]string{"a"},
+			noInstrumentationLibraryName,
+			nil,
+		},
+		{
+			"single dim w/ instrumentation library name and only one label",
+			SingleDimensionRollupOnly,
+			[]string{"a"},
+			"cloudwatch-otel",
+			nil,
+		},
+		{
+			"zero + single dim w/o instrumentation library name",
+			ZeroAndSingleDimensionRollup,
+			[]string{"a", "b", "c"},
+			noInstrumentationLibraryName,
+			[][]string{
+				{},
```

Comment:
@mxiamxia or @shaochengwang: is this the desired output? Do we include `{}` as part of the output if we perform zero dim rollup with no instrumentation library name?

Reply:
yes. {} will be appended to dimensionArray. It indicates `Zero dimension`

Reply:
I see, so what's the effect of `{}` vs just removing it entirely in the exported metrics?

Reply:
if you remove `{}`, we can't generate a metric without dimension attaching which is what we want here

Reply:
Got it, thanks! Does this CR look good to both of you then?

Reply:
Can you add a case for ZeroDimensionRollup for metric with no label (zero dimension)?

Reply:
@shaochengwang done!